pkg/oc/cli/admin/prune/images/images.go

```diff
@@ -344,10 +348,21 @@ func (o PruneImagesOptions) Run() error {
 		limitRangesMap[limit.Namespace] = limits
 	}
 
-	allImages, err := o.ImageClient.Images().List(metav1.ListOptions{})
+	ctx := context.TODO()
+	allImagesUntyped, err := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+		return o.ImageClient.Images().List(opts)
+	}).List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
+	allImages := &imagev1.ImageList{}
```

Comment:
Doesn't this still have the issue that we need to successfully page all of it into memory? And we need each page to load? We should try to produce images async and prune as early as possible. Can we determine, given streams and tags, whether an image needs to be pruned before we have the full list of images?

Reply:
@stevekuznetsov the problem is that we need to pruner layer blobs. Before we delete a blob, we need to be sure that this won't break any image, i.e. the blob can be deleted only when all images that use this blob are going to be pruned. And we can't do it without getting all images.

Reply:
@stevekuznetsov yes we still need it in memory, but we won't suffer a timeout from the apiserver. This PR won't eliminate the risk of prune failing - it will just fail with OOMKilled which will be much less likely to occur.